src/server_manager/web_app/app.ts

```diff
@@ -186,13 +190,54 @@ export class App {
       // Show any manual servers if they exist.
       if (manualServers.length > 0) {
         this.showManualServerIfHealthy(manualServers[0]);
-        return;
+        return Promise.resolve();
       }
 
       // User has no manual servers - check if they are logged into DigitalOcean.
-      const accessToken = this.digitalOceanTokenManager.extractTokenFromUrl();
+      const accessToken = this.digitalOceanTokenManager.getStoredToken();
       if (accessToken) {
-        return this.getDigitalOceanServerList(accessToken)
+        return this.enterDigitalOceanMode(accessToken);
+      }
+
+      // User has no manual servers or DigitalOcean token.
+      this.showIntro();
+      return Promise.resolve();
+    });
+  }
+
+  // Show the DigitalOcean server creator or the existing server, if there's one.
+  private enterDigitalOceanMode(accessToken: string) {
+    const doSession = this.createDigitalOceanSession(accessToken);
+    // We create a <div>, but all we really need is an EventTarget.
+    const events = document.createElement('div');
+    let cancelled = false;
+
+    const query = () => {
+      if (cancelled) {
+        return;
+      }
+      this.digitalOceanRetry(() => doSession.getAccount())
```

Comment:
Do you want this to keep retrying even after cancellation?

Reply:
Good catch. I'm cancelling the retries too now.